server/common/fbs/matrix.py

```diff
@@ -43,81 +38,6 @@ def serialize_matrix(builder, n_rows, n_cols, columns, col_idx):
-    are provided by type_info.
-    """
-
-    arr = source_array
-    (array_type, as_type) = encoding_info(source_array)
-
-    if isinstance(arr, pd.Index):
-        arr = arr.to_series()
-
-    # convert to a simple ndarray
-    if as_type == "json":
-        as_json = arr.to_json(orient="records")
-        arr = np.array(bytearray(as_json, "utf-8"))
-    else:
-        if sparse.issparse(arr):
-            arr = arr.toarray()
-        elif isinstance(arr, pd.Series):
-            arr = arr.to_numpy()
-        if arr.dtype != as_type:
-            arr = arr.astype(as_type)
-
-    # serialize the ndarray into a vector
-    if arr.ndim == 2:
-        if arr.shape[0] == 1:
-            arr = arr[0]
-        elif arr.shape[1] == 1:
-            arr = arr.T[0]
-
-    vec = builder.CreateNumpyVector(arr)
-
-    # serialize the typed array table
-    builder.StartObject(1)
-    builder.PrependUOffsetTRelativeSlot(0, vec, 0)
-    array_value = builder.EndObject()
-    return (array_type, array_value)
-
-
-def column_encoding(arr):
-    column_encoding_type_map = {
-        # array protocol string:  ( array_type, as_type )
-        np.dtype(np.float64).str: (TypedArray.TypedArray.Float32Array, np.float32),
-        np.dtype(np.float32).str: (TypedArray.TypedArray.Float32Array, np.float32),
-        np.dtype(np.float16).str: (TypedArray.TypedArray.Float32Array, np.float32),
-        np.dtype(np.int8).str: (TypedArray.TypedArray.Int32Array, np.int32),
-        np.dtype(np.int16).str: (TypedArray.TypedArray.Int32Array, np.int32),
-        np.dtype(np.int32).str: (TypedArray.TypedArray.Int32Array, np.int32),
-        np.dtype(np.int64).str: (TypedArray.TypedArray.Int32Array, np.int32),
-        np.dtype(np.uint8).str: (TypedArray.TypedArray.Uint32Array, np.uint32),
-        np.dtype(np.uint16).str: (TypedArray.TypedArray.Uint32Array, np.uint32),
-        np.dtype(np.uint32).str: (TypedArray.TypedArray.Uint32Array, np.uint32),
-        np.dtype(np.uint64).str: (TypedArray.TypedArray.Uint32Array, np.uint32),
-    }
-    column_encoding_default = (TypedArray.TypedArray.JSONEncodedArray, "json")
-
-    encoding_dtype = np.dtype(get_encoding_dtype_of_array(arr))
-    return column_encoding_type_map.get(encoding_dtype.str, column_encoding_default)
-
-
-def index_encoding(arr):
-    index_encoding_type_map = {
-        # array protocol string:  ( array_type, as_type )
-        np.dtype(np.int32).str: (TypedArray.TypedArray.Int32Array, np.int32),
-        np.dtype(np.int64).str: (TypedArray.TypedArray.Int32Array, np.int32),
-        np.dtype(np.uint32).str: (TypedArray.TypedArray.Uint32Array, np.uint32),
-        np.dtype(np.uint64).str: (TypedArray.TypedArray.Uint32Array, np.uint32),
-    }
-    index_encoding_default = (TypedArray.TypedArray.JSONEncodedArray, "json")
-
-    return index_encoding_type_map.get(arr.dtype.str, index_encoding_default)
-
-
 def guess_at_mem_needed(matrix):
```

Comment:
It would be easy to finish the implementation of this with [pandas.DataFrame.memory_usage](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.memory_usage.html), and would give you a modest perf improvement.